src/partitioning/qbvh.rs

```diff
@@ -415,6 +415,10 @@ impl<T: IndexedData> QBVH<T> {
         stack.clear();
         stack.push(0);
 
+        if self.nodes.is_empty() {
+            return;
+        }
+
```

Comment:
Let's not do a useless push if there is no node.

```suggestion
        stack.clear();

        if !self.nodes.is_empty() {
            stack.push(0);
        }
```